File: Java/src/main/java/org/codecop/dependencies/parameterise_constructor/Discount.java
package org.codecop.dependencies.parameterise_constructor;

public class Discount {

    private final MarketingCampaign marketingCampaign;

    public Discount() {
        this.marketingCampaign = new MarketingCampaign();
    }

    public Money discountFor(Money netPrice) {
        if (marketingCampaign.isCrazySalesDay()) {
            return netPrice.reduceBy(15);
        }
        if (netPrice.moreThan(Money.ONE_THOUSAND)) {
            return netPrice.reduceBy(10);
        }
        if (netPrice.moreThan(Money.ONE_HOUNDRED) && marketingCampaign.isActive()) {
            return netPrice.reduceBy(5);
        }
        return netPrice;
    }
}

File: CSharp/DependencyBreakingKatas/ParameteriseConstructor/Discount.cs
namespace Org.Codecop.Dependencies.ParameteriseConstructor
{
    public class Discount
    {
        private readonly MarketingCampaign marketingCampaign;

        public Discount()
        {
            this.marketingCampaign = new MarketingCampaign();
        }

        public virtual Money DiscountFor(Money netPrice)
        {
            if (marketingCampaign.IsCrazySalesDay())
            {
                return netPrice.ReduceBy(15);
            }
            if (netPrice.MoreThan(Money.OneThousand))
            {
                return netPrice.ReduceBy(10);
            }
            if (netPrice.MoreThan(Money.OneHoundred) && marketingCampaign.IsActive())
            {
                return netPrice.ReduceBy(5);
            }
            return netPrice;
        }
    }
}
